### PR TITLE
Use PSR-11 container

### PR DESCRIPTION
**Pull request by romm**
The dependency to `container-interop/service-provider:^0.4` changes the
signature of the container, from the container-interop container to
PSR-11 container.

### CODE REVIEW

**Comment by moufmouf** [comment]
Build is failing because of thecodingmachine/discovery (that is a composer plugin and Composer is bumping into 2.0 for its API soon).
I'll need to check/fix that!

**Comment by romm** [comment]
I began the fix in #7, I need some more time but I need to sleep before (I'm currently stuck in New-Zealand because of the Covid, timezone is not the same at all 😄).

File: src/ServiceProviderCompilationPass.php [hunk 1]
```diff
@@ -135,11 +135,12 @@ private function getServiceDefinitionFromCallable($serviceName, $serviceProvider
             // TODO: plug the definition-interop converter here!
         }*/
         $factoryDefinition = new Definition('Class'); // TODO: in PHP7, we can get the return type of the function!
+        $factoryDefinition->setPublic(true);
```
Comment (romm) [inline src/ServiceProviderCompilationPass.php:138]: With more recent Sf versions, definitions have to be public (at least that's what I believe). If you think I missed something, please tell me. 😉

File: src/SymfonyContainerAdapter.php [hunk 2]
```diff
@@ -33,13 +30,8 @@ public function get($id)
         if ($this->container->hasParameter($id)) {
             return $this->container->getParameter($id);
         }
-        try {
```
Comment (romm) [inline src/SymfonyContainerAdapter.php:36]: Symfony container is PSR-11 compliant so no need to adapt exceptions.

Comment (moufmouf) [reply src/SymfonyContainerAdapter.php:36]: True!

File: src/ServiceProviderCompilationPass.php (hunk 1 above)
Comment (moufmouf) [reply src/ServiceProviderCompilationPass.php:138]: I guess it really depends on how you use those services. container-interop/service-provider does not have the notion of public/private loke Symfony does. If your definitions are private, it only means you must refer to those definitions in a constructor, and you won't be able to do something like `$container->get('MyDefinition')` (which is actually rather a good thing).

Of course, there are times where being able to call `$container->get('MyDefinition')` is useful (especially when you want to do lazy loading). The best solution here would maybe be to add a flag to the ServiceProviderBridgeBundle constructor to be able to specify if definitions are public or not.

Comment (moufmouf) [reply src/ServiceProviderCompilationPass.php:138]: Meanwhile, I'm :+1: making this public by default.

**Comment by moufmouf** [comment]
Hey @romm !

Stuck in New Zealand! Woah ;) Hope it's not too hard there.

Thanks a lot for the PR. I'm OK merging this as is,this is clearly an improvement! I think it could be slightly improved by adding an option to choose if services are private or public. See https://github.com/thecodingmachine/service-provider-bridge-bundle/pull/6#discussion_r407894994

Do you feel like adding this before we merge this PR?

File: src/ServiceProviderCompilationPass.php (hunk 1 above)
Comment (romm) [reply src/ServiceProviderCompilationPass.php:138]: I completely agree but there are cases where constructor dependency injection cannot happen (I can give an example if needed), and the fact is that service providers don't have a notion of private/public services; this is only in Symfony (maybe in other frameworks as well but that's not my point).

So I think bothering the user with a flag in the bundle is not really needed, but maybe we'd need an input from someone else. 😊

Comment (moufmouf) [reply src/ServiceProviderCompilationPass.php:138]: Ok, let's merge this then!

**Comment by romm** [comment]
Thanks. 😊

No, life in New Zealand is probably the best one right now (almost no Covid cases), I'm lucky: I am in a farm but have enough internet for some web development during the lockdown. 😁

See you in other PR I guess!

**Pull request merged**